PCSX2/pcsx2
# Capture: Implement audio file customization

### Description of Changes
Gives users the ability to change the channel arrangement (mono or stereo) for a captured wav file.

### Rationale behind Changes
We're most likely leaning towards a complete separation of audio and video capture - or really, allowing for audio to be captured separately from video if a user chooses to do so. It makes sense to thus add some switchable options to expand the use cases.

Also, wavfile.h/cpp from soundtouch had a lot of irrelevant functions for our context, so moving towards a completely custom version of those files that can be used on any pc config is probably best.

### Suggested Testing Steps
1. Check to ensure that pcsx2_ui.ini has these written correctly.
2. Check in some external audio program (audacity pre-3.0 for example) that the wav format matches what settings were chosen.
